lib/LaTeXML/Package/latexml.sty.ltxml

```diff
@@ -329,38 +365,34 @@ DefConstructor('\lxDeclare OptionalMatch:* OptionalKeyVals:Declare {}',
     AssignValue(mathfont => LaTeXML::Common::Font->mathDefault(), 'local');
     return; },
   afterDigest => sub { my ($stomach, $whatsit) = @_;
-    my ($star, $keys, $pattern) = $whatsit->getArgs;
-    return unless $keys;
-    CheckOptions("\\lxDeclare keys", $declare_keys, %{ $keys->getKeyVals });
+    my ($star, $kv, $pattern) = $whatsit->getArgs;
+    return unless $kv;
+    CheckOptions("\\lxDeclare keys", $declare_keys, %{ $kv->getKeyVals });
     foreach my $key (qw(role tag name meaning replace)) {
-      if (my $value = $keys->getValue($key)) {
+      if (my $value = $kv->getValue($key)) {
         Warn('unexpected', $key, $stomach,
           "Repeated $key: " . join('; ', map { Stringify($_) } @$value))
           if ref $value eq 'ARRAY'; } }
-    my $id = ($keys->getValue('tag') ? next_declaration_id() : undef);
+    my $id = ($kv->getValue('tag') || $kv->getValue('description') ? next_declaration_id() : undef);
     if ($id && LookupValue('InPreamble')) {
       Warn('unexpected', 'tag', $stomach,
         "Declaration with tag cannot appear in preamble"
           . Stringify($whatsit)); }
-    if (my $tag = $keys->getValue('tag')) {
-      if (!$keys->getValue('text')) {
-        my ($newtag, $text) = splitDeclareTag($tag);
-        if ($text) {
-          $keys->setValue(tag  => $newtag);
-          $keys->setValue(text => $text);
-    } } }
-    $whatsit->setProperties(scope => getDeclarationScope($keys),
-      role    => ToString($keys->getValue('role')),
-      name    => ToString($keys->getValue('name')),
-      meaning => ToString($keys->getValue('meaning')),
-      trace   => defined $keys->getValue('trace'),
-      wrap    => defined $keys->getValue('wrap'),
-      nowrap  => defined $keys->getValue('nowrap'),
+    # Temporary(?) Hack: If no description, bui tag is of form <math>: text
+    # make description = tag, and tag be only the shorter text part
+    $whatsit->setProperties(scope => getDeclarationScope($kv),
+      role    => ToString($kv->getValue('role')),
+      name    => ToString($kv->getValue('name')),
+      meaning => ToString($kv->getValue('meaning')),
+      trace   => defined $kv->getValue('trace'),
+      wrap    => defined $kv->getValue('wrap'),
+      nowrap  => defined $kv->getValue('nowrap'),
```

Comment:
Likely not immediate feedback, but having `wrap` and `nowrap` as two separate booleans is a trap in hiding. In particular, the case where "somehow" we end up with both `wrap=>1, nowrap=>1`. Maybe it's slightly cleaner with a single attribute with 3 values? `undef`, `1`, `0` for "default", "force", "never".

Reply:
To a certain extent these are left-overs. During development, there kept being cases of automatically wrap this, but don't wrap that, etc.  Was useful to experiment, but I should take a look and cleanup before merging. Between TeX and Perl, reliably maintaining those 3 values can be tricky though!

Reply:
I removed `wrap`, since it was dead code.